docs/design/proposals/destructive-operations/destructive-operations.md

```diff
@@ -0,0 +1,130 @@
+
+(Requirements 2) The full annotation path needs to consist of an annotation prefix (typically a URL) before the annotation name. One variation should concern every ACK controller regardless of service, and one should discriminate for controllers only concerned with a single service. Cross-service annotations, finalizers and CRDS are already configured with the `services.k8s.aws/` prefix. For this option, prepending the service identifier to the beginning would identify that the annotation applies only to the given service, eg. `s3.services.k8s.aws/` . Hence, there would be two prefixes:
+
+* `services.k8s.aws/`: All ACK controllers will read the policy of this namespace/resource
+* `<service>.services.k8s.aws/`: Only the <service> ACK controller will read the policy of this namespace (taking precedence over the former option)
+
+
+The name of this annotation could be either:
+
+* Follow the AWS variations (`RemovalPolicy` or `DeletionPolicy`) which would put it inline with expectations of current CDK and CloudFormation users
+* Follow existing K8s variations (`ResourcePolicy` or `ReclaimPolicy`) which would put it inline with expectations of current K8s users
+
+Users familiar with K8s would find the existing K8s variations more recognizable but may incorrectly assume that the annotation has the same values and behaves identically to native K8s resources. Therefore, it may be better to introduce the AWS terminology into the annotation - as the proposed functionality will behave identically to CDK and CloudFormation. Considering this annotation takes effect after the user calls “Delete”, I propose using `DeletionPolicy` as the annotation name.
+
+(Requirement 3) In order to account for a future possibility of adding additional policy options, the value of this annotation should be an enumeration. Considering the name already references the existing CloudFormation `DeletionPolicy` field, the values should similarly follow those values:
+
+* `Delete` - deletes the AWS resource alongside the CR
+* `Retain` - retains the AWS resource, but deletes the CR
+
+The default value for this annotation will be `Delete` as it is in-line with the existing ACK functionality of deleting the underlying AWS resource.
+
+#### Examples
+
+Definition at the controller - all resources reconciled by this controller will use the “retain” policy (unless overridden by an annotation):
+
+```yaml
+containers:
+    - command:
+    - ./bin/controller
+    args:
+    - --aws-account-id
+    - "$(AWS_ACCOUNT_ID)"
+    - --aws-region
+    - "$(AWS_REGION)"
+    - --enable-development-logging
+    - "$(ACK_ENABLE_DEVELOPMENT_LOGGING)"
+    - --log-level
+    - "$(ACK_LOG_LEVEL)"
+    - --resource-tags
+    - "$(ACK_RESOURCE_TAGS)"
++  - --deletion-policy
++  - "retain"
```

Comment:
For the drift remediation feature, [we went for resource level configuration](https://github.com/aws-controllers-k8s/runtime/pull/106#discussion_r1055738553). On top of the `--deletion-policy` flag, do we want to add another flag to have the similar experience for deletion protection? thinking:
```bash
--resource-deletion-policy Function=Retain \
--resource-deletion-policy EventSourceMappings=Retain
```

Reply:
I think I actually have some issues with the resource level configuration of drift remediation, in that it's a non-standard format and seems a little bit convoluted. But I'll leave those comments on your PR.

For now I'll stick to having a global deletion policy flag, and leave the door open to more granular CLI flags later